app/lib/katello/errors.rb

```diff
@@ -106,6 +106,7 @@ def self.from_exception(exception)
 
     class CandlepinNotRunning < StandardError; end
     class CandlepinPoolGone < CandlepinError; end
+    class CandlepinEnvironmentGone < CandlepinError; end
```

Comment:
Could we have a generic class like `CandlepinResourceGone` that can be used for both pool and environment and all other kinds of resources in Candlepin? So we don't need to update this file again for similar issues from other resources.

Reply:
Let me think about that!

Reply:
We could easily do that, but I don't think there's a value in doing it right now. At the moment, in each case, we want to know specifically that the Environment is not there or the Pool is not there. No use case right now for catching such errors generically, but would be easy to make them extend a generic class as you suggest in the future as they are laid out here.